eZ/Publish/API/Repository/RoleService.php

```diff
@@ -338,6 +343,30 @@ public function assignRoleToUser(Role $role, User $user, RoleLimitation $roleLim
     public function unassignRoleFromUser(Role $role, User $user);
 
     /**
+     * Loads a user group role assignment for the given id.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the role assignment, role or user group was not found
+     *
+     * @param mixed $assignmentId
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\UserGroupRoleAssignment
+     */
+    public function loadUserGroupRoleAssignment($assignmentId);
+
+    /**
+     * Loads a user role assignment for the given id.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the role assignment, role or user was not found
+     *
+     * @param mixed $assignmentId
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\UserRoleAssignment
+     */
+    public function loadUserRoleAssignment($assignmentId);
```

Comment:
@pspanja If you load the wrong type of assignment you get a NotFoundException. Having two methods here saves us from an extra fetch. I could use just one, but then I must first try to load the contentId as a user group, and if that fails, load it as a user. The assignments table has just a contentId, it doesn't specify if it's a group or a user. (We could of course have a loadRoleAssignment() method in addition to these two, so you can be specific when you know what you have, and use autodetection when you don't.)